src/features/lightspeed/vue/views/webviewMessageHandlers.ts

```diff
@@ -499,4 +535,132 @@ export class WebviewMessageHandlers {
+  private async runDevcontainerCreateProcess(
+    payload: DevcontainerFormInterface,
+    webview: vscode.Webview,
+  ) {
+    const { destinationPath, image, isOverwritten } = payload;
+    let commandResult: string;
+    let message: string;
+    let commandOutput = "";
+
+    commandOutput += `------------------------------------ devcontainer generation logs ----------------------------------------\n`;
+
+    const destinationPathUrl = destinationPath;
+
+    const devcontainerExists = fs.existsSync(
+      path.join(expandPath(destinationPathUrl), ".devcontainer"),
+    );
+
+    const imageURL = this.getContainerImage(image);
+    const recommendedExtensions = this.getRecommendedExtensions();
+
+    if (devcontainerExists && !isOverwritten) {
+      message = `Error: Devcontainer already exists at ${destinationPathUrl} and was not overwritten. Use the 'Overwrite' option to overwrite the existing file.`;
+      commandResult = "failed";
+    } else {
+      try {
+        commandResult = await this.createDevcontainer(
+          destinationPathUrl,
+          recommendedExtensions,
+          imageURL,
+        );
+        if (commandResult === "failed") {
+          message =
+            "ERROR: Could not create devcontainer. Please check that your destination path exists and write permissions are configured for it.";
+        } else {
+          message = `Created new devcontainer at ${destinationPathUrl}`;
+        }
+      } catch (error) {
+        commandResult = "failed";
+        message = `ERROR: Could not create devcontainer: ${error}`;
+      }
+    }
+
+    commandOutput += message;
+    console.debug(message);
+
+    webview.postMessage({
+      command: "execution-log",
+      arguments: {
+        commandOutput: commandOutput,
+        projectUrl: destinationPathUrl,
+        status: commandResult,
+      },
+    });
+  }
+
+  private getContainerImage(dropdownImage: string): string {
+    if (dropdownImage.includes("Upstream")) {
+      return DevcontainerImages.Upstream;
+    } else if (dropdownImage.includes("Downstream")) {
+      return DevcontainerImages.Downstream;
+    }
+    return DevcontainerImages.Upstream;
+  }
+
+  private getRecommendedExtensions(): string[] {
+    return DevcontainerRecommendedExtensions.RECOMMENDED_EXTENSIONS;
+  }
+
+  private async createDevcontainer(
+    destinationUrl: string,
+    recommendedExtensions: string[],
+    devcontainerImage: string,
+  ): Promise<string> {
+    try {
+      const expandedPath = expandPath(destinationUrl);
+      const devcontainerDir = path.join(expandedPath, ".devcontainer");
+      const devcontainerJsonPath = path.join(
+        devcontainerDir,
+        "devcontainer.json",
+      );
+
+      // Create .devcontainer directory if it doesn't exist
+      if (!fs.existsSync(devcontainerDir)) {
+        fs.mkdirSync(devcontainerDir, { recursive: true });
+      }
+
+      // Create devcontainer.json content
+      const devcontainerConfig = {
```

Comment:
The devcontainer should be generated from the source files under resources/contentCreator/ for devcontainer. The whole .devcontainer directory should be scaffolded along with the folders/files inside in the structure they're stored in, then the substitutions for image and extensions will go into those template files before creation. 

See the original scaffold function here: https://github.com/ansible/vscode-ansible/pull/2137/files#diff-ecac48deb6c5ffcb0ec4fd219eb62e7dea0a92e108811b3e229ce89f9defbe0cL371